src/OpenTracing/Util/GlobalTracer.cs

```diff
@@ -61,19 +61,21 @@ public sealed class GlobalTracer : ITracer
         /// <returns>Whether a tracer has been registered.</returns>
         public static bool IsRegistered()
         {
-            return s_isRegistered;
+            lock (s_lock)
```

Comment:
I don't think this actually solves anything, at worst it means checking is a Tracer has been registered now incurs taking a lock.

In the example code given in the issue:

```csharp
if (!OpenTracing.Util.GlobalTracer.IsRegistered())
    OpenTracing.Util.GlobalTracer.Register(Tracer);
```

Even if the lock is present, this code is still not thread safe as another thread can take the lock and register a tracer.

Reply:
Are you sure? It seems that since `Register` is calling `RegisterIfAbsent` which is under a lock, only one of them is actually going to get it.

Reply:
The `RegisterIfAbsent` seems fine and should stay, however I don't see the value in including a lock on the `IsRegistered` call.

Reply:
Yeah, now that you mention it, I'm questioning it as well. @MikeGoldsmith, thoughts?

Reply:
I think the original issue may have lead me to add that, on review I too don't think it's necessary.

The benefit it gives is that if a tracer is being registered, a call to see if a tracer has been registered will wait until that action completes (whether successfully or not).

Reply:
Sorry for butting in here, but GlobalTracer is just a singlton if I am not mistaken, so why don't you use Lazy<T> initialization for "s_instance"? You'll get: lazy initialization and you can also use the "isThreadSafe" parameter to make the object thread safe?

Reply:
@MirzaMerdovic `GlobalTracer` isn't really a singleton per-se, as library users can register an implementation of `ITracer` to be accessible at `GlobalTracer.Instance`. I don't see how `Lazy<T>()` could achieve that.

Reply:
True, my mistake haven't figured out the code properly. Lazy instance couldn't ensure thread safety because the "s_isRegistered" would remain unsafe.
Sorry for troubling you guys.

Reply:
@MirzaMerdovic no problem adding your opinion. As a weird side note, a `Lazy<T>` value may be thread-safe, but the initialiser method is not.

I do agree with @johnduhart, a `Lazy<T>` won't help in this circumstance.

I can remove the `lock` around returning the * s_isRegistered*. The only value it adds is it will wait to to return if another thread is actively trying to register a Tracer but with the preferred method of `RegisterIfAbsent`, it doesn't really matter.

Reply:
@austinlparker what is your preference here? Returning * s_isRegistered` with or without the `lock`?

Reply:
I think without the lock makes more sense.